### ISSUE TITLE
CDH 5.3.1 - No container-executor

### ISSUE THREAD

**Issue by kitplummer** [saltstack-formulas/hadoop-formula#21]
I'm not experienced enough with CDH/YARN to know what's really supposed to happen from the distro - but the 'container-executor' binary isn't there in the ${hadoop}/bin directory - or anywhere else.  This causes the state to fail.  Anyone else see this?

```----------
          ID: fix-executor-group
    Function: cmd.run
        Name: chown root /usr/lib/hadoop/bin/container-executor
      Result: False
     Comment: Command "chown root /usr/lib/hadoop/bin/container-executor" run
     Started: 14:33:50.911201
    Duration: 46.19 ms
     Changes:
              ----------
              pid:
                  10772
              retcode:
                  1
              stderr:
                  chown: cannot access `/usr/lib/hadoop/bin/container-executor': No such file or directory
              stdout:

----------
          ID: fix-executor-group
    Function: cmd.run
        Name: chgrp yarn /usr/lib/hadoop/bin/container-executor
      Result: False
     Comment: Command "chgrp yarn /usr/lib/hadoop/bin/container-executor" run
     Started: 14:33:50.957615
    Duration: 48.23 ms
     Changes:
              ----------
              pid:
                  10787
              retcode:
                  1
              stderr:
                  chgrp: cannot access `/usr/lib/hadoop/bin/container-executor': No such file or directory
              stdout:

----------
          ID: fix-executor-permissions
    Function: cmd.run
        Name: chmod 06050 /usr/lib/hadoop/bin/container-executor
      Result: False
     Comment: One or more requisite failed: {'hadoop.yarn.fix-executor-group': 'Command "chgrp yarn /usr/lib/hadoop/bin/container-executor" run'}
     Started:
    Duration:
     Changes:```

**Comment by kitplummer** [comment]
Same issue with CDH-5.0.0.

**Comment by sroegner** [comment]
Looks to me like Cloudera doesn't put the container-executor into their tarballs - it is in that path in all the HDP tarballs i checked. Not sure what can or should be done about this - is moving to one of the HDP versions an option?

**Comment by kitplummer** [comment]
Unfortunately it isn't an option.

I think the only thing would be to install Maven and build it.  Yuck.

**Comment by gravyboat** [comment]
@sroegner @kitplummer Is there anything unique to the Cloudera install we could check against? As much as I dislike it, we could add a pillar value to flag these as to whether we should run them or not based on that pillar value.

**Comment by kitplummer** [comment]
The existence of the binary I suppose.  I'm not sure how necessary it is to
YARN - especially considering it isn't part of the CDH distro.  Maybe just
flag to ignore if CDH-*?

**Comment by gravyboat** [comment]
Hmm, I mean we could just add an unless that looks like this:

```
- unless: test ! -f /usr/lib/hadoop/bin/container-executor
```

That would return true if the file doesn't exist.

**Comment by kitplummer** [comment]
That'd work.  :)

**Comment by gravyboat** [comment]
@kitplummer Works for me!

**Comment by sroegner** [comment]
Good idea. The binary is only needed if yarn is (explicitly) configured to use it.

**Comment by gravyboat** [comment]
@sroegner Were you referring to the unless, or the binary check?

**Comment by sroegner** [comment]
@gravyboat I meant the unless check

**Comment by gravyboat** [comment]
@sroegner Okay cool!

**Comment by gravyboat** [comment]
I'm closing this, it's resolved in #22. Thanks for the input!
